modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosControllerEpics.scala

```diff
@@ -230,47 +229,41 @@ object GmosControllerEpics extends GmosEncoders {
   def setWavelength(state: GmosCCEpicsState, w: Length): Option[IO[Unit]] =
     applyParam(state.disperserWavel, encode(w), CC.setDisperserLambda)
 
-  def setDisperserParams[T <: GmosController.SiteDependentTypes: Encoders](state: GmosCCEpicsState, cfg: GmosController.Config[T], d: GmosController.Config[T]#GmosDisperser): List[Option[IO[Unit]]] = {
+  def setDisperserParams[T <: GmosController.SiteDependentTypes: Encoders](state: GmosCCEpicsState,
+                                                                           cfg: GmosController.Config[T],
+                                                                           d: GmosController.Config[T]#GmosDisperser)
+  : List[Option[IO[Unit]]] = {
     val set: List[Option[IO[Unit]]] = d match {
-      case cfg.GmosDisperser.Mirror =>
-        val s: Option[IO[Unit]] = setDisperser(state, cfg.mirror)
-        //If disperser is set, force mode configuration
-        s.foldMap(_ => List(s, CC.setDisperserMode(disperserMode0).some))
+      case cfg.GmosDisperser.Mirror => List(setDisperser(state, cfg.mirror))
       case cfg.GmosDisperser.Order0(d) =>
         val s0: Option[IO[Unit]] = setDisperser(state, d)
         // If disperser is set, force order configuration
-        val s: List[Option[IO[Unit]]] = if(s0.isEmpty) List(setOrder(state, Order.ZERO))
-                else CC.setDisperserOrder(disperserOrderEncoder.encode(Order.ZERO)).some :: List(s0)
-        //If disperser or order are set, force mode configuration
-        s.foldMap(_ => s ++ List(CC.setDisperserMode(disperserMode0).some))
+        if(s0.isEmpty) List(setOrder(state, Order.ZERO))
```

Comment:
`s0 match ...`?

Reply:
I would have used `match` if I were to use the contents of `s0`. But as it is this is more compact.